test/test_EDM4hepFile.py

```diff
@@ -0,0 +1,508 @@
+    assert dqdx.getTrack() == track
+
+
+def test_GeneratorEventParametersCollection(event, particle):
+    """Check the GeneratorEventParametersCollection"""
+    counter = count()
+    gep_coll = event.get("GeneratorEventParametersCollection")
+    assert len(gep_coll) == 1
+    gep = gep_coll[0]
+    assert gep.getEventScale() == next(counter)
+    assert gep.getAlphaQED() == next(counter)
+    assert gep.getAlphaQCD() == next(counter)
+    assert gep.getSignalProcessId() == next(counter)
+    assert gep.getSqrts() == next(counter)
+    xsecs = gep.getCrossSections()
+    xsec_errors = gep.getCrossSectionErrors()
+    assert len(xsecs) == VECTORSIZE
+    assert len(xsec_errors) == VECTORSIZE
+    for i in range(VECTORSIZE):
+        assert xsecs[i] == next(counter)
+        assert xsec_errors[i] == next(counter)
+
+    assert len(gep.getSignalVertex()) == 1
+    assert gep.getSignalVertex()[0] == particle
+
+
+def test_GeneratorPdfInfoCollection(event):
+    """Check the GeneratorPdfInfoCollection"""
+    counter = count()
+    gpi_coll = event.get("GeneratorPdfInfoCollection")
+    assert len(gpi_coll) == 1
+    gpi = gpi_coll[0]
+    assert gpi.getPartonId(0) == next(counter)
+    assert gpi.getPartonId(1) == next(counter)
+    assert gpi.getLhapdfId(0) == next(counter)
+    assert gpi.getLhapdfId(1) == next(counter)
+    assert gpi.getX(0) == next(counter)
+    assert gpi.getX(1) == next(counter)
+    assert gpi.getXf(0) == next(counter)
+    assert gpi.getXf(1) == next(counter)
+
+    assert gpi.getScale() == next(counter)
+
+
+def check_LinkCollection(event, coll_type, from_el, to_el):
+    """Check a single link collection of a given type"""
+    counter = count()
+    coll_name = f"{coll_type}Collection"
+    link_coll = event.get(coll_name)
+    assert len(link_coll) == 1
+    link = link_coll[0]
+    assert link.getWeight() == next(counter)
+    assert link.getFrom() == from_el
+    assert link.getTo() == to_el
+
+
+def test_LinkCollections(event, particle, reco_particle, track):
+    """Check all the link collections"""
+    calo_hit = event.get("CalorimeterHitCollection")[0]
+    simcalo_hit = event.get("SimCalorimeterHitCollection")[0]
+    cluster = event.get("ClusterCollection")[0]
+    tracker_hit = event.get("TrackerHit3DCollection")[0]
+    simtracker_hit = event.get("SimTrackerHitCollection")[0]
+    vertex = event.get("VertexCollection")[0]
+
+    check_LinkCollection(event, "CaloHitSimCaloHitLink", calo_hit, simcalo_hit)
+    check_LinkCollection(
+        event, "TrackerHitSimTrackerHitLink", tracker_hit, simtracker_hit
+    )
+    check_LinkCollection(event, "CaloHitMCParticleLink", calo_hit, particle)
+    check_LinkCollection(event, "ClusterMCParticleLink", cluster, particle)
+    check_LinkCollection(event, "TrackMCParticleLink", track, particle)
+    check_LinkCollection(event, "RecoMCParticleLink", reco_particle, particle)
+    check_LinkCollection(event, "VertexRecoParticleLink", vertex, reco_particle)
```

Comment:
```suggestion
    check_LinkCollection(event, "RecoMCParticleLink", reco_particle, particle)
    check_LinkCollection(event, "CaloHitSimCaloHitLink", calo_hit, simcalo_hit)
    check_LinkCollection(
        event, "TrackerHitSimTrackerHitLink", tracker_hit, simtracker_hit
    )
    check_LinkCollection(event, "CaloHitMCParticleLink", calo_hit, particle)
    check_LinkCollection(event, "ClusterMCParticleLink", cluster, particle)
    check_LinkCollection(event, "TrackMCParticleLink", track, particle)
    check_LinkCollection(event, "VertexRecoParticleLink", vertex, reco_particle)
```
Just like the rest it's easier to check that nothing is missing if it's sorted like in `edm4hep.yaml`

Reply:
Done.